src/Pact/Types/Lang.hs

```diff
@@ -622,9 +624,23 @@ instance Show Ref where
     show (Direct t) = abbrev t
     show (Ref t) = abbrev t
 
+-- | Gas compute cost unit.
+newtype Gas = Gas Int64
+  deriving (Eq,Ord,Num,Real,Integral,Enum)
+instance Show Gas where show (Gas g) = show g
+instance Monoid Gas where
+  mempty = 0
+  (Gas a) `mappend` (Gas b) = Gas $ a + b
+
+
+-- | Price per 'Gas' unit.
+newtype GasPrice = GasPrice Decimal
```

Comment:
It's a bit surprising to me that `GasPrice` is a decimal, so rounding will be a factor. Though I don't see it being used yet so it's hard to tell how this will work.